### DISCUSSION TITLE
Verständnisfrage zu InvoiceDescriptor.Load(filename)

### DISCUSSION THREAD

**Discussion by vieledinge** [stephanstapel/ZUGFeRD-csharp#328] [Q&A]
Hallo allerseits,

ich spiele gerade mit dem Projekt, wobei mich primär das Lesen von Rechnungen interessiert. Scheitere aber aktuell daran, dass mir bei allen Dateien eine Exception um die Ohren fliegt (Kein ZUGFeRD-Reader kann den Typ parsen), obwohl es sich dabei auch um ZUGFeRD-PDFs handelt, die auch eine saubere Typdeklaration (ZUGFeRD 2.2, Factur-X, XRechnung) im XML haben.

Wo ich nicht ganz schlau draus werde: Erwartet InvoiceDescriptor.Load(filename) eine PDF-Datei mit eingebetteten XML-Daten oder die schon extrahierte XML-Datei?

Und falls eine PDF-Datei erwartet wird:  Wenn ich hier eine PDF-Datei ohne ZUGFeRD-Daten übergebe, bekomme ich dann als Profile UNKNOWN zurück oder wird eher eine Exception geworfen (Keine ZUGFeRD-Rechnung).

VG
Uwe

**Comment by stephanstapel** [comment]
Hallo @vieledinge ,

es wird die extrahierte xml-Datei erwartet.
Hier ist eine Beschreibung, wie dies z.B. implementiert werden kann:

https://github.com/stephanstapel/ZUGFeRD-csharp?tab=readme-ov-file#extracting-xml-attachments-from-pdf-files

mir ist leider keine open source pdf-handling-Bibliothek bekannt.

**Reply by vieledinge** [reply]
Danke, das hilft mir schon mal weiter. Hab ich ja dann Montag was zu tun....
